pkg/storage/storage.go

```diff
@@ -169,6 +169,7 @@ type AuthorizationModelBackend interface {
 type StoresBackend interface {
 	CreateStore(ctx context.Context, store *openfgapb.Store) (*openfgapb.Store, error)
 	DeleteStore(ctx context.Context, id string) error
+	UpdateStore(ctx context.Context, id string, name string) (*openfgapb.Store, error)
```

Comment:
If we're going to introduce a very specific storage method just for updating the store name, then let's at least introduce it in a way that can extend to other fields potentially that may get added as metadata to a store in the future. Extending the storage interface 1:1 for entity/field combo is not in the best interest of the maintainability of the storage interface long term.

```go
package storage

type StoresBackend interface {
    UpdateStore(ctx context.Context, req *openfgapb.UpdateStoreRequest) (*openfgapb.Store, error)
}
```

Another option would be to make the storage interface more generic. Essentially all we have is a Write and Read. Updating or creating a store is a Write operation (put based semantics with the Store entity provided).